website/packages/docs/src/pages/css-extraction-webpack.mdx

```diff
@@ -1,49 +1,145 @@
 ---
 section: 50-Guides
-name: Webpack CSS Extraction
+name: Stylesheet extraction
 ---
 
-# Webpack CSS Extraction
+# Stylesheet Extraction
 
-Extracting to a static style sheet can be enabled with some extra configuration.
-We recommend only extracting when building an application for production.
+Stylesheet extraction is **highly recommended** for all component libraries and applications, so that generated class names can be combined and de-duplicated into one external stylesheet. This allows you to enjoy the decreased stylesheet size that comes with atomic styling.
 
-If you're building a design system or component library and shipping to NPM you shouldn't extract just yet,
-let your consumers do it in their app.
+There are two types of stylesheet extraction:
 
-## Setup
+- Platform stylesheet extraction
+- Parcel / Webpack stylesheet extraction
 
-Before continuing make sure to install the [webpack loader](/installation#webpack).
+## Which type should I use?
 
-### Compiled configuration
+If your package is a component library, use our **platform stylesheet extraction**. This includes Atlassian Design System packages like `@atlaskit/button`; or any other package that isn't run directly, but is imported and used by _another codebase_ (e.g. Jira, Confluence).
 
-Configure the `extract` option to `true` in the webpack loader and turn on the extract plugin.
+If your package exists in the same codebase as, and is _directly consumed_ by, a Webpack or Parcel app, use **Parcel stylesheet extraction** or **Webpack stylesheet extraction** depending on what bundler your app uses.
 
-```diff
-// webpack.config.js
-+const { CompiledExtractPlugin } = require('@compiled/webpack-loader');
+Note that Parcel and Webpack are the only bundlers we support.
+
+## Platform stylesheet extraction
+
+Platform stylesheet extraction only works if you use Babel to build your component library. Note that we do not recommend nor support using bundlers (e.g. Webpack or Parcel) to build your component library, because bundlers are appropriate for web apps (e.g. Jira) and not for component libraries.
+
+> **Warning**<br/>
+> Ensure that any apps that import your component library also have the Compiled Webpack or Parcel plugins installed (`@compiled/webpack-loader` or `@compiled/parcel-config`), otherwise the order in which the styles are applied will become unpredictable. This can be an issue if you have styles that overlap, e.g. if you mix `padding` and `paddingTop` in the same component, or have `@media` queries that potentially overlap.<br/><br/>
+> If the apps that import your component library use other bundlers like Vite or Next.js, or if your apps cannot install the Compiled Webpack or Parcel plugins, stylesheet extraction will still work! However, you must make sure that within each component, you should _never_ mix any CSS properties or styles that overlap.
```

Comment:
This will certainly fall on deaf ears, but hey, it's in here!

Reply:
🫠